fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -212,6 +212,33 @@ where
             PowerUpdates::default()
         };
 
+        // check for upgrades in the upgrade_scheduler
+        match &self.upgrade_scheduler {
+            Some(upgrade_scheduler) => {
+                let chain_id: u64 = state.chain_id().into();
+                let block_height: u64 = state.block_height().try_into().unwrap();
+
+                if let Some(upgrade) = upgrade_scheduler.get(chain_id, block_height) {
+                    tracing::info!(
+                        chain_id = chain_id,
+                        height = block_height,
+                        "Running migration",
```

Comment:
Maybe a migration title would be informative in the future. Or a more formal application version. 

That is actually part of what is returned by the `info` method in the `app.rs`. There are `version` and `app_version` there, both hardcoded at the moment:
```rust
    /// The application software semantic version.
    pub version: String,
    /// The application protocol version.
    pub app_version: u64,
```

I thought `version` is only about the `fendermint` binary, and `app_version` is what we are migrating here. It could be interesting to give some structure to `app_version` with shifts, e.g. an IPC version and a Fluence version could be packed in it with bit shifting, although I'm not sure if CometBFT expects some ordering on it. Or we can just surrender control over it to these upgrades, after all it would be the operator who applies them, they have to opt into IPC upgrades as much as their own. 

In any case, it would be something to think about: to return not just `PowerUpdates` but a potential new app version as well from the interpreter. Then it would also be clear in which method the upgrades are applied, if it was part of the `EndOutput`, and then `App` would know to save it somewhere.

Looking at the `FvmStateParams::network_version`, I reckon `app_version` belongs there too, and it's not clear what the difference between the two is exactly. `App::info` does have access to it at least, and I think if you ask what the app version was in a previous height, it should report the one before the migration, so it's part of the state.

Reply:
If you add a field for it, this might just be the time to start worrying about backwards compatibility. I should have added a golden file for `FvmStateParams` CBOR, but I doubt that I did :(